### DISCUSSION TITLE
Expose custom error property

### DISCUSSION THREAD

**Discussion by nephix** [elysiajs/elysia#836] [Q&A] [answered]
Hi elysians 👋 

I skimmed through the docs trying to find a way to expose a property of my custom `Error` that I'm using but didn't manage to find a way yet

```ts
class BaseError extends Error {
    readonly extensions: { [key: string]: string };

    constructor(message: string, code: string) {
        super(message);
        this.extensions = { code };
    }
}

export enum CustomErrorCodes {
    SIGN_IN_REQUIRED = 'You are not signed in.',
}

type CustomErrorCode = keyof typeof CustomErrorCodes;
export class CustomError extends BaseError {
    public constructor(public code: CustomErrorCode) {
        super(CustomErrorCodes[code], code);
    }
}
```

When I then throw this error eg in a `.get()`:

```ts
.get('/test') => {
    throw new CustomError('SIGN_IN_REQUIRED');
})
```

 I only get the `message`:

```json
{"name":"Error","message":"You are not signed in."}
```

Is there an easy way to also expose the `code`? I would expect something like:

```json
{"name":"Error","message":"You are not signed in.","extensions":{"code":"SIGN_IN_REQUIRED"}}
```

**Comment by kravetsone** [comment]
you can do everything what u want with onError hook

**Reply by nephix** [reply]
How would I type the `error` from `onError()` to my `CustomError` in order to access `extensions.code`?

Ok I guess something like
```ts
.onError(({ error }) => {
    if (error instanceof BaseError) {
        return error.extensions.code;
    }
})
```

works just hoped there was a way around `instanceof`

**Reply by kravetsone** [reply]
> How would I type the `error` from `onError()` to my `CustomError` in order to access `extensions.code`?
> 
> Ok I guess something like
> ```ts
> .onError(({ error }) => {
>     if (error instanceof BaseError) {
>         return error.extensions.code;
>     }
> })
> ```
> 
> works just hoped there was a way around `instanceof`

Use . error method

**Reply by nephix** [reply]
Ah I didn't know about `.error()` is this documented somewhere? All the pages give me 404 when I search
Or a quick example

**Reply by kravetsone** [reply, marked as answer]
> Ah I didn't know about `.error()` is this documented somewhere? All the pages give me 404 when I search
> Or a quick example

https://elysiajs.com/essential/life-cycle.html#custom-error

**Reply by nephix** [reply]
Thanks a bunch @kravetsone
